### PR TITLE
Created a function "validateUserCredentials()" to validate the user credentials during config set, check-node, prep-node execution.

### PR DESCRIPTION
**Pull request by AnirudhPokala**
Screenshot 2021-03-01 at 6 58 58 PM
Screenshot 2021-03-01 at 6 58 17 PM

### CODE REVIEW

File: cmd/checkNode.go [hunk 1]
```diff
@@ -61,3 +60,27 @@ func checkNodeRun(cmd *cobra.Command, args []string) {
...
+	}
+
+	_, err = c.Keystone.GetAuth(
+		ctx.Username,
+		ctx.Password,
+		ctx.Tenant,
+	)
+
+	if err != nil {
+		zap.S().Fatalf("Invalid Credentials (Username/Password/Service), run 'pf9ctl config set' !!")
```
Comment (nikhilbandi) [inline cmd/checkNode.go:83]: Better error message is: "Invalid credentials (Username/ Password/ Account), run 'pf9ctl config set' with correct credentials. "

Comment (AnirudhPokala) [reply cmd/checkNode.go]: Done Nikhil edited the error message.

File: cmd/checkNode.go [hunk 2]
```diff
@@ -32,19 +38,12 @@ func init() {
...
-	executor, err := getExecutor()
-	if err != nil {
-		zap.S().Fatalf("Error connecting to host %s", err.Error())
-	}
-	c, err := pmk.NewClient(ctx.Fqdn, executor, ctx.AllowInsecure, false)
-	if err != nil {
-		zap.S().Fatalf("Unable to load clients needed for the Cmd. Error: %s", err.Error())
-	}
+	// Validate the user credentials entered during config set and will bail out if invalid
+	c = validateUserCredentials(ctx)
```
Comment (anupbarve) [inline cmd/checkNode.go:46]: I don't like the idea of a validation function returning a client and operation flow using it. It is non-intuitive. We should pass the client to the `validateUserCredentials`.

Comment (AnirudhPokala) [reply cmd/checkNode.go:46]: Ok anup will create a seperate function to create the client and will pass it to validateUserCredentials. Is it ok?

File: cmd/checkNode.go [hunk 3]
```diff
@@ -61,3 +63,32 @@ func checkNodeRun(cmd *cobra.Command, args []string) {
...
+	_, err = c.Keystone.GetAuth(
+		ctx.Username,
+		ctx.Password,
+		ctx.Tenant,
+	)
+
+	if err != nil {
+		zap.S().Fatalf("Invalid credentials (Username/ Password/ Account), run 'pf9ctl config set' with correct credentials.")
+	}
+}
```
Comment (anupbarve) [inline cmd/checkNode.go:79]: We should move this to config.go

File: cmd/checkNode.go [hunk 4]
```diff
@@ -61,3 +63,32 @@ func checkNodeRun(cmd *cobra.Command, args []string) {
...
+	if err != nil {
+		zap.S().Fatalf("Error connecting to host %s", err.Error())
+	}
+	c, err := pmk.NewClient(ctx.Fqdn, executor, ctx.AllowInsecure, false)
+	if err != nil {
+		zap.S().Fatalf("Unable to load clients needed for the Cmd. Error: %s", err.Error())
+	}
+
+	return c
+}
```
Comment (anupbarve) [inline cmd/checkNode.go:94]: This should not be a different function at all. We should simply unfold it and call it in respective cmd/<command.go> files.

**Review by anupbarve — CHANGES_REQUESTED** [review]

File: cmd/checkNode.go (hunk 3 above)
Comment (AnirudhPokala) [reply cmd/checkNode.go]: sure anup done.

File: cmd/checkNode.go (hunk 4 above)
Comment (AnirudhPokala) [reply cmd/checkNode.go]: ok anup done once review it.

**Review by anupbarve — APPROVED** [review]
Please take help from @devidask27 to run TC build and post the TC link here.

File: cmd/config.go [hunk 5]
```diff
@@ -72,3 +97,17 @@ func init() {
 	configCmdCreate.AddCommand(configCmdGet)
 	configCmdCreate.AddCommand(configCmdSet)
 }
+
+// This function will validate the user credentials entered during config set and bail out if invalid
+func validateUserCredentials(pmk.Config, pmk.Client) {
```
Comment (sahil-lakhwani) [inline cmd/config.go:102]: I think this function should return error instead of failing in a fatal way.

Comment (AnirudhPokala) [reply cmd/config.go:102]: Ya we can do that, but then we are adding same error message line thirce in config.go, checkNode.go, prepNode.go. So added that error validation in function itself.

**Comment by AnirudhPokala** [comment]
> Please take help from @devidask27 to run TC build and post the TC link here.

Ya Anup its working fine. 

http://teamcity.platform9.horse:8111/buildConfiguration/Pf9project_Platform9Components_Pf9ctl/1452392

File: cmd/config.go (hunk 5 above)
Comment (sahil-lakhwani) [reply cmd/config.go:102]: I think that should be fine. This function can return an error message, which the callers can use however they want

Comment (AnirudhPokala) [reply cmd/config.go]: Ya Sahil modified it, once review it.

**Review by sahil-lakhwani — APPROVED** [review]

**Pull request merged**